Tools/clinic/clinic.py

```diff
@@ -5735,21 +5730,23 @@ def main(argv: list[str]) -> None:
         return
 
     if not ns.filename:
-        cmdline.print_usage()
-        sys.exit(-1)
+        parser.error("no input files")
 
     if ns.output and len(ns.filename) > 1:
-        print("Usage error: can't use -o with multiple filenames.")
-        print()
-        cmdline.print_usage()
-        sys.exit(-1)
+        parser.error("can't use -o with multiple filenames")
 
     for filename in ns.filename:
         if ns.verbose:
             print(filename)
         parse_file(filename, output=ns.output, verify=not ns.force)
 
 
-if __name__ == "__main__":
-    main(sys.argv[1:])
+def main(argv: list[str] | None = None) -> NoReturn:
+    parser = create_cli()
+    args = parser.parse_args(argv)
+    run_clinic(parser, args)
     sys.exit(0)
```

Comment:
IMO, `sys.exit(0)` as the last line of a command-line utility feels like writing C code in Python :)

Reply:
Heh, maybe. The motivation is to make the calling `clinic.main()` programmatically have ~exactly the same behaviour as calling `python Tool/clinic/clinic.py` on the command line (now they both raise `SystemExit` at the end, whether or not they fail). That means that we can have a nice symmetry between `expect_failure()` and `expect_success()` in the test suite

Reply:
Of course, that's a very good reason to use `sys.exit(0)`!